src/System.Linq.Expressions/tests/Call/CallTests.cs

```diff
@@ -82,6 +82,19 @@ public static void ArrayWriteBack(bool useInterpreter)
             Assert.Equal(2, lambda(array));
         }
 
+        [Theory]
+        [ClassData(typeof(CompilationTypes))]
+        public static void EnumArgAndReturn(bool useInterpreter)
+        {
```

Comment:
Maybe add tests for all other argument counts?
Also, can we test that the cache is actually being used?

Reply:
We have lots of tests that exercise the new code paths. I just needed to check what happens in a case of Enum (since TypeCode returns the underlying type) and found that none of the tests are touching that scenario. So I have added one scenario with Enum.

Otherwise, I think the coverage for the new code is very good. It does handle the most common scenarios.

I am not sure the caches are easily testable though. I think the only kind of regression that can happen is if they are not used at all. If they work incorrectly we will see lots of failures (it did happen more than once in development).

I think I am comfortable with the testing the caches indirectly. 
There are many other caches in the interpreter/compiler that we do not directly test.

---
In reply to: [181906548](https://github.com/dotnet/corefx/pull/28792#discussion_r181906548) [](ancestors = 181906548)